### PR TITLE
feat(connector cluster): add update command

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Add connector cluster update command with flags to change `name` and `annotations`.

### Verification Steps

1. Build the CLI locally
```
make binary
```
3. Run the connector cluster update command:
```
./rhoas connector cluster update --id <cluster-id> --name test-rename --annotations="environment=production,app=nginx" -v
```

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

File: pkg/cmd/connector/cluster/update/update.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	httpRes, err := a.Execute()
+	if httpRes != nil {
+		defer httpRes.Body.Close()
+	}
+
+	if apiErr := connectorerror.GetAPIError(err); apiErr != nil {
+		if apiErr.GetCode() == connectorerror.ERROR_7 {
+			return opts.f.Localizer.MustLocalizeError("connector.cluster.update.error.notFound", localize.NewEntry("ID", opts.id))
+		}
```
Comment (jackdelahunt) [inline pkg/cmd/connector/cluster/update/update.go:98]: I believe there should be more errors here to catch [here](https://api.openshift.com/?urls.primaryName=connector%20service%20fleet%20manager%20service#/Connector%20Clusters/updateConnectorClusterById)

**Review by jackdelahunt — COMMENTED** [review]
Looks good I think we just need more error handling for this

**Comment by dhirajsb** [comment]
functionally lgtm

**Review by jackdelahunt — APPROVED** [review]
LGTM

**Comment by rkpattnaik780** [comment]
> functionally lgtm

Thank you.

**Pull request merged**